pkgs/checks/lib/src/extensions/async.dart

```diff
@@ -131,15 +186,148 @@ extension StreamChecks<T> on Check<StreamQueue<T>> {
     });
   }
 
-  /// Expects that the `Stream` closes without emitting any even that satisfies
+  /// Expects that the stream satisfies each condition in [conditions] serially.
+  ///
+  /// Waits for each condition to be satisfied or rejected before checking the
+  /// next. Subsequent conditions will not see any events consumed by earlier
+  /// conditions.
+  ///
+  /// ```dart
+  /// await checkThat(StreamQueue(someStream)).inOrder([
+  ///   it()..emits().that(it()..equals(0)),
+  ///   it()..emits().that(it()..equals(1)),
+  //  ]);
+  /// ```
+  ///
+  /// If this expectation fails, the source queue will be left in its original
+  /// state.
+  /// If this expectation succeeds, consumes as many events from the source
+  /// stream as are consumed by all the conditions.
+  Future<void> inOrder(Iterable<Condition<StreamQueue<T>>> conditions) async {
+    conditions = conditions.toList();
+    final descriptions = <String>[];
+    await _expectAsync(
+        () => descriptions.isEmpty
+            ? ['satisfies ${conditions.length} conditions in order']
+            : descriptions, (actual) async {
+      var satisfiedCount = 0;
+      for (var condition in conditions) {
+        descriptions.addAll(await describeAsync(condition));
+        final failure = await softCheckAsync(actual, condition);
+        if (failure != null) {
+          final which = failure.rejection.which;
+          return Rejection(actual: [
+            'a stream'
+          ], which: [
+            if (satisfiedCount > 0)
+              'satisfied ${satisfiedCount} conditions then',
+            'failed to satisfy the condition at index ${satisfiedCount}',
+            if (failure.detail.depth > 0) ...[
+              'because it:',
+              ...indent(
+                  failure.detail.actual.skip(1), failure.detail.depth - 1),
+              ...indent(prefixFirst('Actual: ', failure.rejection.actual),
+                  failure.detail.depth),
+              if (which != null)
+                ...indent(prefixFirst('Which: ', which), failure.detail.depth),
+            ] else ...[
+              if (which != null) ...prefixFirst('because it ', which),
+            ],
+          ]);
+        }
+        satisfiedCount++;
+      }
+      return null;
+    });
+  }
+
+  /// Expects that the stream statisfies at least one condition from
+  /// [conditions].
+  ///
+  /// If this expectation fails, the source queue will be left in its original
+  /// state.
+  /// If this expectation succeeds, consumes the same events from the source
+  /// queue as the satisfied condition. If multiple conditions are satisfied,
+  /// chooses the condition which consumed the most events.
+  Future<void> anyOf(Iterable<Condition<StreamQueue<T>>> conditions) async {
```

Comment:
This has the same name as https://github.com/dart-lang/test/blob/7157541a93fd7217355f1e8323de7745751a12bd/pkgs/checks/lib/src/extensions/core.dart#L58 and is now relying on the specificity of the generic to be chosen for stream queues instead of the synchronous one. I was initially avoiding this, but it's one of the advantages of extension methods, so I think it's fine to lean on it a bit. WDYT?

Reply:
I think its probably fine, jump to definition etc are very reliable for extension methods